### DISCUSSION TITLE
Need help with NiDAQmx plugin

### DISCUSSION THREAD

**Discussion by aurore-finco** [PyMoDAQ/PyMoDAQ#88] [Q&A]
Dear Pymodaq friends,

I am currently working on a Pymodaq module to perform optically detected magnetic resonance measurements. My aim is to end up with a 1D viewer recording the spectra, inheriting from a daq_move handling the microwave source and one or several daqmx objects handling the NI card which controls the measurement. For the moment, I struggle with the pymodaq daqmx plugin. Maybe you can give me some hints to start.

I need to :
- Send a trigger signal (with a digital output actuator I guess) to my microwave source at every point of the spectrum to trigger the next frequency.
- Count the number of photons coming during a given amount of time right after the trigger signal has been sent (with a counter detector? a digital input detector? both?)
- Also record an analog input voltage during the same counting time.

First question, do I need a detector/actuator for each channel of the card that I will use, since I need to specify a "Signal type"?

Then, how do I handle the synchronization between the signals on the different channels? I am used to have also a clock channel, how does it work with the pymodaq module?

For the counting itself, in our previous code we had a digital input as photon source and a counter channel connected to the photon source. Does it work the same here?

Thanks a lot for your help!
Aurore

**Comment by seb5g** [comment]
Hello this should probably go into the dedicated repo that is pymodaq_plugins_daqmx. 

However here are some hints concerning your questions. The daqmx plugins are just general features usable as a plugin, however for more complex dealing with triggers/clocks/digital... I would recommend to use the high level object DAQmx object and related object from the repo:

![image](https://user-images.githubusercontent.com/6609955/205106000-1d5dedc4-717e-4db8-9aef-f6b848627c3c.png)

Then inside your acquisition (completely custom) plugin, you'll declare channels, clocks, triggers and configure tasks and start them. You should have a look at this implementation (in my custom pymodaq_plugins_moke) see below

![image](https://user-images.githubusercontent.com/6609955/205105852-f4b459be-f164-4150-acfc-6d64692f75ee.png)

where here I record several analog signal, process them before exporting data.
Let me know if this is unclear

**Reply by aurore-finco** [reply]
Sorry for posting on the bad repo, I was not sure if this had to be posted as an issue.
Thanks a lot for your answer, I will look into that.
